experimental/cuda2/status_util.h

```diff
@@ -0,0 +1,107 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_EXPERIMENTAL_CUDA2_STATUS_UTIL_H_
+#define IREE_EXPERIMENTAL_CUDA2_STATUS_UTIL_H_
+
+#include <stdint.h>
+
+#include "experimental/cuda2/dynamic_symbols.h"
+#include "iree/base/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+//===----------------------------------------------------------------------===//
+// CUDA result macros
+//===----------------------------------------------------------------------===//
+
+// Converts a CUresult to an iree_status_t.
+//
+// Usage:
+//   iree_status_t status = CU_RESULT_TO_STATUS(cuda_symbols, cuDoThing(...));
+#define CU_RESULT_TO_STATUS(syms, expr, ...) \
```

Comment:
would be nice to IREE_ prefix all of these macros to avoid conflicts/confusion as to where they come from
(if these were only ever going to be internal less of an issue, but with the nvgpu stuff some of these headers are leaking out even though it's YMMV - at least naming them will make it easier for users to know where they come from and decide if they want to use IREE code vs something supported by CUDA itself)

Reply:
Good points. Great for clarity and consistency too.